src/envoy/http/authn/origin_authenticator.cc

```diff
@@ -92,7 +92,7 @@ bool OriginAuthenticator::run(Payload* payload) {
 
   // If none of the JWT triggered, origin authentication will be ignored, as if
   // it is not defined.
-  return !triggered || success;
+  return (policy_.origins_size() > 0 && !triggered) || success;
```

Comment:
I think we can simplify the switch into 
```
if (policy_.origins_size() == 0 && policy_.principal_binding()==iaapi::PrincipalBinding::USE_ORIGIN) {
  // This should be rejected by validation during policy submitting. Log warning in case validation was
  // was not run properly.
  ENVOY_LOG(warn,
                   "Principal is binded to origin, but no method specified in "	                   "Principal is binded to origin, but no method specified in "
                   "policy {}",	                   "policy {}",
                   policy_.DebugString());	                   policy_.DebugString());
  return false;
}
```

And this return value is

```
// Always returns successful (true) if binding is USE_PEER, else 
// returns true if no jwt was triggered, or triggered and success.
policy_.principal_binding() == iaapi::PrincipalBinding::USE_PEER || !triggered || success;
```
(you may also move and rename the `success` variable to `triggered_success` to have clearer semantic)

Reply:
It looks like a change that have different behaviors, is it okay?
When binding is `USE_PEER` but origins not empty, before it may return false if origins validate fail. After (now) it always return true.

Reply:
Ah, I miss read the code :). It's then even simpler, `return !triggered || success` should be equivalent, right?

Reply:
There is still a difference: 
When origins is empty and binding is `USE_PEER`, the success will be set to true in turn the setOriginResult() and setPrintipal() will be called.

How about just change the switch case to `return false` early and include the following logic in an else statement? (see my latest update)

Reply:
My comment including change of the switch statement to `if` and return early. See the first comment for snippet.

Reply:
Just to confirm one last thing, when binding is `USE_ORIGIN`:
If origins is empty and , we return false
If origins is non empty but none of them triggered, we return true
This is the desired behavior, right?

Reply:
Yes, given non triggered == non matching given path.

Reply:
Great! Could you approve if no more changes needed? Thanks.